### ISSUE TITLE
Improve docs regarding error reporting under asgi; add FAQ item

### ISSUE THREAD

**Issue by 0x1618** [falconry/falcon#2393]
Hello, I feel like i am missing something because falcon/uvicorn doesn't show any error traceback when there's a server error. Why is that? Just to note, i've used Flask for almost five years and am now migrating to falcon, so i have never used before uvicorn/falcon.

```python
import falcon
import falcon.asgi


class ThingsResource:
    async def on_get(self, req, resp):
        raise ValueError('foo')


app = falcon.asgi.App()

things = ThingsResource()
app.add_route('/things', things)
```

```
uvicorn main:app --reload --log-level debug
```
```
INFO:     Will watch for changes in these directories: ['falcon-server']
INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
INFO:     Started reloader process [6088] using StatReload
INFO:     Started server process [15700]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     127.0.0.1:59567 - "GET /things HTTP/1.1" 500 Internal Server Error # No traceback.
```

**Comment by vytas7** [comment]
Hi @0x1618! No worries, there are no stupid questions, just stupid answers :slightly_smiling_face:

The issue here is that ASGI doesn't define anything like WSGI's [`wsgi.errors`](https://peps.python.org/pep-3333/#environ-variables), so there is no standardized way or logging a traceback back to the app server.

This issue has already been reported before (#2098), but I see that we only added a [section in the tutorial](https://falcon.readthedocs.io/en/stable/user/tutorial-asgi.html#debugging-asgi-applications), which is maybe not that easy to discover. We ought to add a FAQ item too, possibly linking to the tutorial. Also edit that section, explaining the difference between ASGI and WSGI in this case.

TL;DR @0x1618 you need to configure the `falcon` logger using the stdlib's logging module, either directly or just configure the root logger:
```python
import logging

import falcon
import falcon.asgi

logging.basicConfig(
    format="%(asctime)s [%(levelname)s] %(message)s", level=logging.INFO)


class ThingsResource:
    async def on_get(self, req, resp):
        raise ValueError('foo')


app = falcon.asgi.App()

things = ThingsResource()
app.add_route('/things', things)
```

You should now see a traceback being logged:
```
2024-10-28<...> [ERROR] [FALCON] Unhandled exception in ASGI app
Traceback (most recent call last):
  File <... venv ...>, line 524, in __call__
    await responder(req, resp, **params)
  File "/tmp/test.py", line 12, in on_get
    raise ValueError('foo')
ValueError: foo
```

**Comment by CaselIT** [comment]
thanks for the feedback! I think @vytas7 wanted to keep this issue open as a reminding to improve the docs on this

**Comment by vytas7** [comment]
Yeah, we already have that section in the tutorial, but it is not very obvious, I would like to highlight it in the FAQ too.
